### ISSUE TITLE
how to use it ?

### ISSUE THREAD

**Issue by Ranper** [ahhussein/SciIE-GraphWriter#2]
Hello author, I am looking at the source code of this paper, but I found that the code of that paper is relatively obscure and uses tensorflow. I am not very familiar with it. Recently, I found the code for your pytorch implementation on github. I feel Very good, it seems to have found a companion. One problem I don’t quite understand is that in addition to sentences and doc_id, the data set dev also contains clusters, ner, relations and other information, and these additional information models are also used in the verification [according to My understanding]. **I was wondering whether the model can extract the entities in the sentence, the relationship between entities and the common reference resolution when only inputting the sentence? ?** Hope to get your reply, thanks!

**Comment by ahhussein** [comment]
Hi Ranper, thanks for showing interest in my work. My code basically converts SciIE into PyTorch with some additional tweaks and further integration steps with a graph to text model that you might choose to discard. As you said, the model can extract entities and relationships while taking a sentence as an input, in other words both entities and relation extraction modules are sentence-level tasks. However, coreference resolution module works on a document level meaning that 2 entities from 2 different sentences can still refer to each other. Nevertheless, you don't have to sample your batches as documents, you can still sample sentences and the coreference resolution will apply to all sentence in the batch and apply a mask to make sure that only entities in a document are detected as coreferences. I am attaching a diagram that depicts the full model architecture, hope it helps.
![image](https://user-images.githubusercontent.com/13599165/98363164-ccb60f00-202e-11eb-8e8b-402971c47fe2.png)
